src/main/java/squid/Squid.java

```diff
@@ -0,0 +1,353 @@
+package squid;
+
+import java.util.Scanner;
+
+import squid.constants.CorrectUsage;
+import squid.constants.Exceptions;
+import squid.constants.Messages;
+import squid.constants.Regex;
+import squid.exceptions.DuplicateTaskNameException;
+import squid.exceptions.IncorrectIndexException;
+import squid.exceptions.IncorrectInputException;
+import squid.exceptions.NotEnoughDatesException;
+import squid.exceptions.NotEnoughInputsException;
+import squid.exceptions.ParseFailException;
+import squid.exceptions.SquidDateException;
+import squid.exceptions.SquidException;
+import squid.tasks.DateTime;
+import squid.tasks.Deadline;
+import squid.tasks.Event;
+import squid.tasks.Task;
+import squid.tasks.Tasks;
+import squid.tasks.Todo;
+
+/**
+ * The main class of Squid.
+ */
+public class Squid {
+
+
+    public Squid() {
+        new Tasks();
+    }
+
+    /**
+     * List all existing tasks.
+     *
+     * @throws NotEnoughInputsException If there are not enough parameters.
+     * @throws IncorrectIndexException Should never happen.
+     */
+    private static void list() throws NotEnoughInputsException, IncorrectIndexException {
+        echo(Messages.LIST);
+        Tasks.list();
+    }
+
+    /**
+     * Initializes Squid.
+     */
+    private static void hello() {
+        try {
+            Tasks.read();
+        } catch (ParseFailException | DuplicateTaskNameException | SquidDateException e) {
+            echo(e.toString());
+        }
+        System.out.println(Messages.LINE_BREAK);
+        echo(Messages.HELLO);
+        System.out.println(Messages.LINE_BREAK);
+    }
+
+    /**
+     * Terminates the program.
+     *
+     * @throws NotEnoughInputsException Should never happen, unless constant MESSAGES.BYE is blank.
+     */
+    private static void bye() throws NotEnoughInputsException {
+        echo(Messages.BYE);
+        Tasks.save();
+    }
+
+    /**
+     * Overloaded method to include this as a valid command.
+     *
+     * @param message The raw input from the user, or message to be printed.
+     * @param isFromUser Whether to further process message.
+     * @throws NotEnoughInputsException If there are not enough parameters.
+     */
+    private static void echo(String message, boolean isFromUser) throws NotEnoughInputsException {
+        if (!isFromUser) {
+            echo(message);
+            return;
+        }
+        String[] params = message.split(" ", 2);
+        if (params.length <= 1) {
+            throw new NotEnoughInputsException(
+                    String.format(
+                            Exceptions.NOT_ENOUGH_INPUTS, "echo", CorrectUsage.ECHO));
```

Comment:
Is there a reason why line 85 is separated from line 84 as compared to line 264?